client/apps/game/src/ui/modules/ws-chat/chat.tsx

```
@@ -623,6 +631,29 @@ function ChatModule() {
     URL.revokeObjectURL(url);
   }, [filteredMessages, directMessageRecipientId, activeRoomId, onlineUsers, offlineUsers, availableRooms]);
 
+  // Add new memoized computations for user lists
+  const usersWithUnreadMessages = useMemo(() => {
+    return [...filteredUsers, ...filteredOfflineUsers]
+      .filter((user) => user && user?.id !== userId && (unreadMessages[user?.id] || 0) > 0)
+      .sort((a, b) => (a?.username || a?.id).localeCompare(b?.username || b?.id));
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)